board/designa/snappermx6/ksz9896.c

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2024 DesignA Electronics Ltd
+ */
+
+#include <common.h>
+#include <linux/errno.h>
+#include <i2c.h>
+
+ #define KSZ9896_MMDACR                                         0x0D
+ #define KSZ9896_MMDAADR                                        0x0E
+
+  //PHY MMD Setup register
+ #define KSZ9896_MMDACR_FUNC                                    0xC000
+ #define KSZ9896_MMDACR_FUNC_ADDR                               0x0000
+ #define KSZ9896_MMDACR_FUNC_DATA_NO_POST_INC                   0x4000
+ #define KSZ9896_MMDACR_FUNC_DATA_POST_INC_RW                   0x8000
+ #define KSZ9896_MMDACR_FUNC_DATA_POST_INC_W                    0xC000
+ #define KSZ9896_MMDACR_DEVAD                                   0x001F
+
+// PHY status register
+#define KSZ9896_PHY_BASIC_STATUS								0x0102
+#define KSZ9896_PHY_LINK_PARTNER_STATUS							0x010A
+#define KSZ9896_PHY_1000BT_STATUS								0x0114
+
+ //MMD LED Mode register
+ #define KSZ9896_MMD_LED_MODE_LED_MODE                          0x0010
+ #define KSZ9896_MMD_LED_MODE_LED_MODE_TRI_COLOR_DUAL           0x0000
+ #define KSZ9896_MMD_LED_MODE_LED_MODE_SINGLE                   0x0010
+ #define KSZ9896_MMD_LED_MODE_RESERVED                          0x000F
+ #define KSZ9896_MMD_LED_MODE_RESERVED_DEFAULT                  0x0001
+  
+ //MMD EEE Advertisement register
+ #define KSZ9896_MMD_EEE_ADV_1000BT_EEE_EN                      0x0004
+ #define KSZ9896_MMD_EEE_ADV_100BT_EEE_EN                       0x0002
+
+
+ //KSZ9896 MMD registers
+ #define KSZ9896_MMD_LED_MODE                                   0x02, 0x00
+ #define KSZ9896_MMD_EEE_ADV                                    0x07, 0x3C
+
+static int ksz9896_write16(struct udevice *dev, uint8_t port, uint16_t reg, uint16_t val)
```

Comment:
This driver should probably be in drivers/net/phy

Reply:
Yes, that is true. I was waiting to see what the Linux .dts file ended up looking like and then replicating it in U-Boot, but I should have started with it there.

Reply:
OK